src/app.js

```diff
@@ -93,21 +95,34 @@ i18next
 
 app.use(i18nextMiddleware.handle(i18next));
 
+// Generate a new session ID asynchronously if needed
+// `express-session` does not support sync session ID generation
+// https://github.com/expressjs/session/issues/107
+app.use(async (req, res, next) => {
+  const sessionId = getCookie(req, SESSION_COOKIE_NAME, SESSION_SECRET);
+  if (!sessionId) {
+    req.generatedSessionId = await uid(24);
```

Comment:
The default behaviour for `express-session` uses `uid.sync` which uses blocking crypto; this version is async.